src/app/clusters/sample-mei-server/sample-mei-server.cpp

```diff
@@ -0,0 +1,228 @@
+#include "sample-mei-server.h"
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Commands.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/InteractionModelEngine.h>
+#include <app/reporting/reporting.h>
+#include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/util.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::SampleMei;
+using namespace chip::app::Clusters::SampleMei::Commands;
+using namespace chip::app::Clusters::SampleMei::Attributes;
+
+SampleMeiServer sampleMeiServer;
+
+// *****************************************************************************
+// Init/Shutdown Callbacks
+
+void MatterSampleMeiPluginServerInitCallback()
+{
+    ChipLogProgress(Zcl, "Sample Mei Init. Ep# %d Total Ep# %u", EMBER_AF_SAMPLE_MEI_CLUSTER_SERVER_ENDPOINT_COUNT,
+                    static_cast<uint16_t>(kNumSupportedEndpoints));
+    ReturnOnFailure(InteractionModelEngine::GetInstance()->RegisterCommandHandler(&sampleMeiServer));
+    VerifyOrReturn(registerAttributeAccessOverride(&sampleMeiServer), CHIP_ERROR_INCORRECT_STATE);
+
+    // registers the fixed endpoints
+    for (uint16_t endpointIndex = 0; endpointIndex < FIXED_ENDPOINT_COUNT; endpointIndex++)
+    {
+        auto endpoint = emberAfEndpointFromIndex(endpointIndex);
+        ChipLogProgress(Zcl, "Creating cluster, Ep index %d", endpoint);
+        if (emberAfContainsServer(endpoint, SampleMei::Id))
+        {
+            sampleMeiServer.RegisterEndpoint(endpoint);
+        }
+    }
+}
+
+void MatterSampleMeiClusterServerShutdownCallback(chip::EndpointId endpoint)
```

Comment:
I doubt this is getting called, since this cluster is not listed under `ClustersWithShutdownFunctions` in `src/app/common/templates/config-data.yaml`.  Was this part tested?

Reply:
Added both to `ClustersWithShutdownFunctions` and `ClustersWithInitFunctions` into `config-data.yaml`. Created https://github.com/project-chip/zap/issues/1134 due to the fact the cluster naming format is different depending on each tag in `config-data.yaml`